zaza/openstack/charm_tests/mysql/tests.py

```diff
@@ -398,3 +400,31 @@ def test_100_kill_crm_master(self):
         # always true.
         assert generic_utils.is_port_open("3306", self.vip), \
             "Cannot connect to vip"
+
+
+class MySQLInnoDBClusterTests(MySQLTest):
+    """Base for mysql-innodb-cluster charm tests."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Run class setup for running mysql-innodb-cluster tests."""
+        super().setUpClass()
+        cls.application = "mysql-innodb-cluster"
+        cls.services = ["mysqld"]
+
+    def test_801_cluster_status(self):
+        """Checking cluster status.
+
+        Run the cluster-status action.
+        """
+        # Update which node is the leader and which are not
+        self.update_leaders_and_non_leaders()
```

Comment:
This is being run for its side-effect of updating the class vars leader/non-leaders ... it would be "better" if the function just returned `(leader, non_leaders)` as a tuple and test functions just called the helper method directly.  i.e. `cls.leader` is just a global in disguise, and it's being updated, which means that its hard to track the side-effects.